### ISSUE TITLE
Fix duplicate songs from being added to user's Spotify playlist

### ISSUE THREAD

**Issue by HendryKaak** [JasonZigelbaum/jqbx-issues#151]
I found out that when I add music from the JQBX web interface to my Spotify lists
and that selected song already exists in that list, it still adds it to the Spotify list.
This might be the same for the JQBX app (correct me if I'm wrong).
 
My suggestion is therefore to *not* add that song to the list or even better, to ask 
if users still wants the song added to the list. Since the developers do have write access 
to Spotify user's list, they probably also have read permissions to read on the user's list
to accomplish that. My guess is that with the read access it is possible to check duplicate
song ID's(not names) by requesting(or simply searching for) the same song from the list
the user wants to add it to.

So, to summarise all this. 
1) The user requests a specific song X to be added to list Y.
2) First check if song ID X exists in list Y with: 
https://developer.spotify.com/documentation/web-api/reference/playlists/get-playlists-tracks/
3) Handle the response by checking if it:
3a) Does exist? Ask the user if it's OK to add song X to list Y even though it's duplicate.
3b) Does not exist? Simply add song X to list Y.

The Spotify app kindly asks the same when duplicate songs are added to the lists.

Edit 1: It seems that the Android app does not (yet?) have the functionality to add song to Spotify user's lists. At least, I could not find it anywhere.

**Comment by Yaouza** [comment]
Good idea, at the moment i use a dedupicator app to find and remove the dupes every months or so..

**Comment by HendryKaak** [comment]
> Good idea, at the moment i use a dedupicator app to find and remove the dupes every months or so..

@Yaouza Thanks for sharing, that is a handy quick fix for now at least.

**Comment by Yaouza** [comment]
> > Good idea, at the moment i use a dedupicator app to find and remove the dupes every months or so..
> 
> @Yaouza Thanks for sharing, that is a handy quick fix for now at least.

Check this out, it's the one i use, it's pretty good -> https://jmperezperez.com/spotify-dedup/
